src/SpectatorView.Unity/Assets/SpatialAlignment.ASA/Scripts/SpatialAlignment/SpatialAnchorsCoordinate.cs

```diff
@@ -23,6 +26,26 @@ internal class SpatialAnchorsCoordinate : SpatialCoordinateUnityBase<string>
         /// <inheritdoc/>
         public override LocatedState State => LocatedState.Tracking;
 
+        private Matrix4x4 CoordinateTransform
+        {
+            get
+            {
+#if UNITY_ANDROID || UNITY_IOS
+                if (Camera.main == null)
```

Comment:
I think there have always been weird perf limitations assocaited with directly accessing Camera.main. If this is gonna be used a bunch (i.e. every frame) it might be worth caching Camera.main in a local var 

https://docs.unity3d.com/ScriptReference/Camera-main.html

'This property uses FindGameObjectsWithTag internally and doesn't cache the result. It is advised to cache the return value of Camera.main if it is used multiple times per frame.'

Reply:
Plus, it appears this check is only needed to be done once.

---
In reply to: [366091123](https://github.com/microsoft/MixedReality-SpectatorView/pull/317#discussion_r366091123) [](ancestors = 366091123)

Reply:
Cached the camera and added some comments.